src/box/tuple.h

```diff
@@ -882,17 +883,10 @@ tuple_next_with_type(struct tuple_iterator *it, enum mp_type type)
 static inline int
 tuple_next_u32(struct tuple_iterator *it, uint32_t *out)
 {
-	uint32_t fieldno = it->fieldno;
 	const char *field = tuple_next_with_type(it, MP_UINT);
 	if (field == NULL)
 		return -1;
 	uint32_t val = mp_decode_uint(&field);
-	if (val > UINT32_MAX) {
```

Comment:
It looks like a bug. `mp_decode_uint()` returns `uint64_t`, which was silently truncated. And still is. File an issue, or fix in the same patchset, please. Better with a test, if you want to fix it by yourself.